torch/csrc/utils/tensor_new.cpp

```diff
@@ -277,7 +276,11 @@ Tensor internal_new_from_data(
 
   auto sizes = compute_sizes(data);
   ScalarType inferred_scalar_type = type_inference ? infer_scalar_type(data) : scalar_type;
-  auto tensor = autograd::make_variable(at::empty(sizes, at::initialTensorOptions().dtype(inferred_scalar_type).pinned_memory(pin_memory)), /*requires_grad=*/false);
+  // This exists to prevent us from tracing the call to empty().  The actual
+  // autograd code doesn't really matter, because requires_grad is always false
+  // here.
+  at::AutoNonVariableTypeMode guard;
+  auto tensor = at::empty(sizes, at::initialTensorOptions().dtype(inferred_scalar_type).pinned_memory(pin_memory));
```

Comment:
should we write
```cpp
at::Tensor tensor = ([&]() {
  at::AutoNonVariableTypeMode guard;
  return at::empty(sizes, at::initialTensorOptions().dtype(inferred_scalar_type).pinned_memory(pin_memory));
}
```
or does the `at::AutoNonVariableTypeMode` guard apply to the rest of code as well?

Reply:
It applies to the rest of the code. At the end of the day I want this entire tensor to come out with NO tracing state applied to it, so it gets handled as a constant.